tests/test_selector.py

```diff
@@ -712,6 +712,39 @@ def test_replacement_null_char_from_body(self):
         self.assertEqual(u'<html><body><p>Grainy</p></body></html>',
                           self.sscls(text).extract())
 
+    def test_characters_gt_and_lt(self):
+        """HTML5 parser tests: greater and less than symbols work as expected."""
+        lt_elem = '20 < 100'
+        gt_elem = '120 > 100'
+        body = u'''<html>
+                    <head></head>
+                    <body>
+                     <div id="distance">{0}</div>
+                    <body>
+                </html>'''
+
+        sel = self.sscls(text=body.format(lt_elem), type='html5')
+        lt_res = sel.xpath('//div[@id="distance"]/text()').get()
+        self.assertEqual(lt_res, lt_elem, msg='less than(<) parsing does not work as expected')
+
+        sel = self.sscls(text=body.format(gt_elem), type='html5')
+        gt_res = sel.xpath('//div[@id="distance"]/text()').get()
+        self.assertEqual(gt_res, gt_elem, msg='greater than(>) parsing does not work as expected')
+
+    def test_complete_tags(self):
+        """HTML5 parser complete/fill tags as expected."""
+        body = u'''<html>
+                    <head></head>
+                       <body>
+                        <li>one<div></li>
+                        <li>two</li>
+                       </body>
+                </html>'''
+        sel = self.sscls(text=body, type='html5')
+        res = sel.xpath('//div/text()').get()
```

Comment:
shouldn't the test confirm that it gets the `div` element? Or maybe I am not understanding the new functionality correctly.

Reply:
Yes, you're right the testcase isn't completely clear. The difference that this test pretend to show is the difference in parsers view. In this particular case:
```
In [1]:         body = u'''<html> 
   ...:                     <head></head> 
   ...:                        <body> 
   ...:                         <li>one<div></li> 
   ...:                         <li>two</li> 
   ...:                        </body> 
   ...:                 </html>'''                                                                                                                                                                                  

In [2]: from parsel import Selector                                                                                                                                                                                 

In [3]: sel = Selector(body, 'html')                                                                                                                                                                                

In [4]: sel.get()                                                                                                                                                                                                   
Out[4]: '<html>\n            <head></head>\n               <body>\n                <li>one<div>\n                <li>two</li>\n               </div></li></body>\n        </html>'

In [5]: selfive = Selector(body, 'html5')                                                                                                                                                                           

In [6]: selfive.get()                                                                                                                                                                                               
Out[6]: '<html><head></head>\n               <body>\n                <li>one<div></div></li>\n                <li>two</li>\n               \n        </body></html>'
```
As you can see, `div` tag is filled/completed/closed in different way with `html` than `html5parser`, just as extra info, the way `html5parser` see is the way that browsers would see that html.

```
In [7]: sel.xpath('//div').get()                                                                                                                                                                                    
Out[7]: '<div>\n                <li>two</li>\n               </div>'

In [8]: selfive.xpath('//div').get()                                                                                                                                                                                
Out[8]: '<div></div>'
```